### ISSUE TITLE
Do not include google classes in jar file

### ISSUE THREAD

**Issue by lowellk** [mttkay/droid-fu#19]
I noticed that the jar you provide has some classes which can also be found in the google guava library. I think it's not good to package these inside your jar because it means trouble for anyone using guava.

**Comment by mttkay** [comment]
Yes, I totally agree. The reason I initially didn't want to depend on the Guava JAR is that it's _massive_ (1.1MB), and it doesn't seem to come in modules, and I only use it for MapMaker.

In other words, declaring a dependency on Guava is fine for building Droid-Fu, but it forces Droid-Fu users to re-distribute the Guava JAR with their app. If they're not using ProGuard, this will inflate their APK by over 1MB. That's a no-go.

I'm open for suggestions though.

**Comment by lowellk** [comment]
How about distributing a second jar which only contains the classes you borrowed from Google. Users can choose to either use that second jar or guava.

**Comment by mttkay** [comment]
I suppose we could do that, but that would be a lot of hassle, since we'd either need it as an artifact on Maven Central or in a custom repository, or have the users install it manually into their local Maven cache before building Droid-Fu.

**Comment by lowellk** [comment]
How does the jarfile get onto github? Can the second jar not be generated similarly?

**Comment by mttkay** [comment]
I uploaded it manually, it's really old actually. For now I assume people will clone the repository and build using Maven to stay up to date. There is no release roadmap at the moment, and I actually have a few things that I'd like to change about Droid-Fu which are fairly fundamental, so at some point in the near future I'm gonna tag the master branch to keep existing users happy and start a version that will make an actual release, but will not be backwards compatible.

I suppose what you suggest is the best solution for now. I'll give it some thought how to make it easy to maintain.

**Comment by lowellk** [comment]
OK, thanks. I may make my own build some day. I've been trying to avoid using maven.

**Comment by mttkay** [comment]
in theory it would be as simple as running 'mvn install', but ever since we added BetterMapActivity, we need the maps.jar to build, which is _not_ on Maven Central (it's not open source). That means you have to use the maven-android-sdk-deployer or install it into your local Maven cache manually.

**Comment by tacone** [comment]
Just got bitten by this and lost a fair bunch of time. I agree to strip google classes from the source :)
